OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -104,6 +113,10 @@ public void GiveResource(int amount)
 			else
 				amount = playerResources.ChangeCash(amount);
 
+			var notifiers = self.World.ActorsWithTrait<INotifyResourceDumped>().Where(x => x.Actor.Owner == self.Owner).Select(x => x.Trait);
```

Comment:
This can then change to
```csharp
foreach (var notify in self.World.ActorsWithTrait<INotifyResourceAccepted>())
{
	if (notify.Actor.Owner != self.Owner)
		continue;

	notify.Trait.OnResourceAccepted(notify.Actor, self, amount);
}
```

The LINQ isn't very helpful here, and calling the interface with the refinery as `self` was wrong.